packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -688,9 +763,17 @@ class BorderDirectional extends BoxBorder {
         bottom.style != topStyle)
       return false;
 
+    if (_strokeAlignIsUniform == false)
+      return false;
+
     return true;
   }
 
+  bool get _strokeAlignIsUniform {
+    final StrokeAlign topStrokeAlign = top.strokeAlign;
+    return start.strokeAlign == topStrokeAlign && bottom.strokeAlign == topStrokeAlign && end.strokeAlign == topStrokeAlign;
```

Comment:
```suggestion
    return start.strokeAlign == topStrokeAlign
      && bottom.strokeAlign == topStrokeAlign
      && end.strokeAlign == topStrokeAlign;
```

Reply:
That also works. My original idea was to make it similar to the others 😅. Fixed.
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/351125/169192354-8578650e-a3ed-4b91-ae1a-28104c95282c.png">

Reply:
It is funny you prefer `&&` at the start than end. The file has both patterns, so it doesn't matter, I'll use exactly the way you wrote. But first time I see someone doing that.

Reply:
We do it because with four spaces before the `&&`, and one after, it makes all of the terms line up under the return. It's not something we apply religiously, only in cases where there are a lot of conditions stacked up.